### PR TITLE
Update config-builder flags and README for clarity

### PR DESCRIPTION
**Pull request by agclement**
## Motivation

The config-builder flag descriptions and README file are not clear for external users.  At the root of this tension is a number of terms (node specifically) that are overloaded with different implicit meaning and a description that is deeply tied to the internal testing use cases.

This PR is an attempt to clarify the flag documentation and README based on an outsider's experience using the config-builder to assist in setting up a TestNet.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Integration tests should suffice.  The change is cosmetic and based on documentation.

## Related PRs

stand alone PR.

### CODE REVIEW

File: config/README.md [hunk 1]
```diff
@@ -38,33 +38,35 @@ configuration from many of the services.
...
 
 ## Generating a new TestNet
 
-The only requirements for generating a TestNet configuration are: (i) having IP
-addresses and ports for each Libra node; (ii) a pre-agreed upon shared secret;
-and (iii) a fixed ordering for each Node in the network. Full node networks can
-either be added to existing configs or generate completely new configs.
+The only requirements for generating the configuration for a full TestNet are: (i) 
+having IP addresses and ports for each Validator; (ii) a pre-agreed upon shared secret
+for the Validator network; and (iii) a fixed ordering for each Validators in the
```
Comment (davidiw) [inline config/README.md:53]: for each Validator (not the lack of s)

File: config/README.md [hunk 2]
```diff
@@ -38,33 +38,35 @@ configuration from many of the services.
...
 
-The only requirements for generating a TestNet configuration are: (i) having IP
-addresses and ports for each Libra node; (ii) a pre-agreed upon shared secret;
-and (iii) a fixed ordering for each Node in the network. Full node networks can
-either be added to existing configs or generate completely new configs.
+The only requirements for generating the configuration for a full TestNet are: (i) 
+having IP addresses and ports for each Validator; (ii) a pre-agreed upon shared secret
+for the Validator network; and (iii) a fixed ordering for each Validators in the 
+network. FullNode configs can either be added to existing configs or generated from 
+scratch.
```
Comment (davidiw) [inline config/README.md:55]: minor nit, would avoid potential colloquialisms like "from scratch" and intead

configs can be generated or added to existing configs

File: config/README.md [hunk 3]
```diff
@@ -38,33 +38,35 @@ configuration from many of the services.
...
-`config-builder` builds an entire network's or swarm's configuration including the
-genesis blob. It takes as one of its input parameters an index that selects, which
-of these configs to return. This can be used to create a Libra TestNet by specifying
-distinct indices for each validator. Similarly the tool can be used to add Fullnodes
-to an existing network.  Finally, it enables generation of a mint/faucet client
-capable of performing mint transactions/creating accounts.
+`config-builder` builds an entire configuration for a Validator or FullNode, 
+including the genesis blob. It takes as one of its input parameters an index that
+specifies the specific node config to return. This can be used to create a Libra 
+TestNet by constructing compatible configurations the full set of Validators.
```
Comment (davidiw) [inline config/README.md:44]: missing a preposition after configurations and before the full set 

could imagine for or across

File: config/README.md [hunk 4]
```diff
@@ -182,3 +199,4 @@ The test configs currently live in `src/config/test_data`.
   interface, AC gRPC)
 - Generating public networks is broken
 - Make SafetyRule use on disk by default and remove from config generator
+
```
[the remark this answers is not in the document]
Comment (agclement) [reply config/README.md]: What is the command to run the linter locally?

I thought it as cargo check or cargo xclippy, but apparently there is another?

File: config/README.md (hunk 1 above)
Comment (agclement) [reply config/README.md]: done.

File: config/README.md (hunk 3 above)
Comment (agclement) [reply config/README.md:47]: done.

**Pull request closed without merging**